src/modules/decentralized-web-node/services/vc-storage.service.ts

```diff
@@ -0,0 +1,78 @@
+import { ConflictException, Injectable } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { FindManyOptions, Repository } from 'typeorm';
+import { Logger } from '../../logger/logger.service';
+import { DataSort } from '../dtos/collections-query-descriptor.dto';
+import { Message } from '../entities/message.entity';
+
+@Injectable()
+export class MessageService {
+  constructor(
+    private readonly logger: Logger,
+    @InjectRepository(Message)
+    private readonly messageRepository: Repository<Message>
+  ) {
+    this.logger.setContext(MessageService.name);
+  }
+
+  async find(
+    user: string,
+    options: {
+      dateSort?: DataSort;
+      dataFormat?: string;
+      schema?: string;
+      objectId?: string;
+    }
+  ): Promise<Message[]> {
+    const selectParams: FindManyOptions<Message> = {
+      where: {
+        createdBy: user,
+        dataFormat: options.dataFormat,
+        schema: options.schema,
+        objectId: options.objectId,
+      },
+    };
+
+    switch (options.dateSort) {
+      case 'createdAscending':
+        selectParams.order = {
+          dateCreated: 'ASC',
+        };
+        break;
+      case 'createdDescending':
+        selectParams.order = {
+          dateCreated: 'DESC',
+        };
+        break;
+      case 'publishedAscending':
+        selectParams.order = {
+          datePublished: 'ASC',
+        };
+        break;
+      case 'publishedDescending':
+        selectParams.order = {
+          datePublished: 'DESC',
+        };
+        break;
+    }
+
+    const items = await this.messageRepository.find(selectParams);
+    return items;
+  }
+
+  async save(message: Message): Promise<Message> {
+    const foundedMessage = await this.messageRepository.findOne({
+      where: {
+        objectId: message.objectId,
+      },
+    });
+    if (foundedMessage) {
+      throw new ConflictException(
+        `Message with id ${message.objectId} already exists`
+      );
+    }
+
+    const item = await this.messageRepository.save(message);
+    return item;
+  }
```

Comment:
I think this logic is incorrect. I think it is possible in principle to update an object. I think rather the `dateCreated` of the previous message should be checked and the object should be updated is the `dateCreated` of the message is newer. https://identity.foundation/decentralized-web-node/spec/#processing-instructions